// @daytonaio/api-client/src/models/sandbox-state-updated-post-request-data.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SandboxStateUpdatedPostRequestDataNewStateEnum = exports.SandboxStateUpdatedPostRequestDataOldStateEnum = void 0;
exports.SandboxStateUpdatedPostRequestDataOldStateEnum = {
    CREATING: 'creating',
    RESTORING: 'restoring',
    DESTROYED: 'destroyed',
    DESTROYING: 'destroying',
    STARTED: 'started',
    STOPPED: 'stopped',
    STARTING: 'starting',
    STOPPING: 'stopping',
    ERROR: 'error',
    BUILD_FAILED: 'build_failed',
    PENDING_BUILD: 'pending_build',
    BUILDING_SNAPSHOT: 'building_snapshot',
    UNKNOWN: 'unknown',
    PULLING_SNAPSHOT: 'pulling_snapshot',
    ARCHIVED: 'archived',
    ARCHIVING: 'archiving',
};
exports.SandboxStateUpdatedPostRequestDataNewStateEnum = {
    CREATING: 'creating',
    RESTORING: 'restoring',
    DESTROYED: 'destroyed',
    DESTROYING: 'destroying',
    STARTED: 'started',
    STOPPED: 'stopped',
    STARTING: 'starting',
    STOPPING: 'stopping',
    ERROR: 'error',
    BUILD_FAILED: 'build_failed',
    PENDING_BUILD: 'pending_build',
    BUILDING_SNAPSHOT: 'building_snapshot',
    UNKNOWN: 'unknown',
    PULLING_SNAPSHOT: 'pulling_snapshot',
    ARCHIVED: 'archived',
    ARCHIVING: 'archiving',
};
