// @daytonaio/api-client/src/models/snapshot-state-updated-post-request-data.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SnapshotStateUpdatedPostRequestDataNewStateEnum = exports.SnapshotStateUpdatedPostRequestDataOldStateEnum = void 0;
exports.SnapshotStateUpdatedPostRequestDataOldStateEnum = {
    BUILD_PENDING: 'build_pending',
    BUILDING: 'building',
    PENDING: 'pending',
    PULLING: 'pulling',
    PENDING_VALIDATION: 'pending_validation',
    VALIDATING: 'validating',
    ACTIVE: 'active',
    INACTIVE: 'inactive',
    ERROR: 'error',
    BUILD_FAILED: 'build_failed',
    REMOVING: 'removing',
};
exports.SnapshotStateUpdatedPostRequestDataNewStateEnum = {
    BUILD_PENDING: 'build_pending',
    BUILDING: 'building',
    PENDING: 'pending',
    PULLING: 'pulling',
    PENDING_VALIDATION: 'pending_validation',
    VALIDATING: 'validating',
    ACTIVE: 'active',
    INACTIVE: 'inactive',
    ERROR: 'error',
    BUILD_FAILED: 'build_failed',
    REMOVING: 'removing',
};
