// @daytonaio/api-client/src/models/volume-state-updated-post-request-data.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Daytona
 * Daytona AI platform API Docs
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@daytona.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeStateUpdatedPostRequestDataNewStateEnum = exports.VolumeStateUpdatedPostRequestDataOldStateEnum = void 0;
exports.VolumeStateUpdatedPostRequestDataOldStateEnum = {
    CREATING: 'creating',
    READY: 'ready',
    PENDING_CREATE: 'pending_create',
    PENDING_DELETE: 'pending_delete',
    DELETING: 'deleting',
    DELETED: 'deleted',
    ERROR: 'error',
};
exports.VolumeStateUpdatedPostRequestDataNewStateEnum = {
    CREATING: 'creating',
    READY: 'ready',
    PENDING_CREATE: 'pending_create',
    PENDING_DELETE: 'pending_delete',
    DELETING: 'deleting',
    DELETED: 'deleted',
    ERROR: 'error',
};
